### ISSUE TITLE
More continue instructions on master

### ISSUE THREAD

**Issue by TheMarex** [Project-OSRM/osrm-backend#2426]
Using this route `http://127.0.0.1:5000/route/v1/driving/13.411474227905273,52.52786686274604;13.424413204193115,52.52342816902858?overview=false&alternatives=true&steps=true` on `master` yields two `continue` instructions that I currently don't see on the demo server yet.

Old:

```
{
    "code": "Ok",
    "routes": [
        {
            "legs": [
                {
                    "steps": [
                        {
                            "geometry": "cjr_IylzpAcAgA",
                            "maneuver": {
                                "bearing_after": 33,
                                "location": [
                                    13.411491,
                                    52.527859
                                ],
                                "bearing_before": 0,
                                "type": "depart"
                            },
                            "duration": 9.2,
                            "distance": 45.1,
                            "name": "Zolastra\u00dfe",
                            "mode": "driving"
                        },
                        {
                            "geometry": "glr_IaozpARgAfBwJF]DOh@yAHSf@aADGR_@",
                            "maneuver": {
                                "bearing_after": 114,
                                "type": "end of road",
                                "modifier": "right",
                                "bearing_before": 33,
                                "location": [
                                    13.411851,
                                    52.5282
                                ]
                            },
                            "duration": 22,
                            "distance": 282.4,
                            "name": "Torstra\u00dfe",
                            "mode": "driving"
                        },
                        {
                            "geometry": "gdr_Ige{pA\\s@N[BENa@Zw@`@oA^gABIFUH[^sAJ_@Nq@Ps@DQRq@Ng@^sADKHYb@kALa@@CT{@Ri@Z_AJ[Lk@\\}Al@sCp@uCH_@He@H]BKJg@TkA",
                            "maneuver": {
                                "exit": 7,
                                "bearing_after": 135,
                                "type": "new name",
                                "modifier": "straight",
                                "bearing_before": 136,
                                "location": [
                                    13.415404,
                                    52.526921
                                ]
                            },
                            "duration": 57,
                            "distance": 725,
                            "name": "Mollstra\u00dfe",
                            "mode": "driving"
                        },
                        {
                            "geometry": "snq_Iw}|pA",
                            "maneuver": {
                                "bearing_after": 0,
                                "location": [
                                    13.424439,
                                    52.523461
                                ],
                                "bearing_before": 116,
                                "type": "arrive"
                            },
                            "duration": 0,
                            "distance": 0,
                            "name": "Mollstra\u00dfe",
                            "mode": "driving"
                        }
                    ],
                    "summary": "Torstra\u00dfe, Mollstra\u00dfe",
                    "duration": 88.2,
                    "distance": 1052.4
                }
            ],
            "duration": 88.2,
            "distance": 1052.4
        }
    ],
    "waypoints": [
        {
            "hint": "H1tMg-RbTIM_wSMAGAAAAFUAAAAAAAAAAAAAAPwUFgptFRYKOLIAAKOkzADzgiEDkqTMAPqCIQMAAAEBMEHR8w==",
            "name": "Zolastra\u00dfe",
            "location": [
                13.411491,
                52.527859
            ]
        },
        {
            "hint": "fCXyg____3_wlgEAFAAAADMAAAAAAAAANgAAAPWs8gPTq_IDOLIAADfXzADFcSEDHdfMAKRxIQMAAAEBMEHR8w==",
            "name": "Mollstra\u00dfe",
            "location": [
                13.424439,
                52.523461
            ]
        }
    ]
}

```

New:

```
{
    "code": "Ok",
    "routes": [
        {
            "legs": [
                {
                    "steps": [
                        {
                            "intersections": [
                                {
                                    "out": 0,
                                    "entry": [
                                        true
                                    ],
                                    "bearings": [
                                        33
                                    ],
                                    "location": [
                                        13.411491,
                                        52.527859
                                    ]
                                }
                            ],
                            "geometry": "cjr_IylzpAcAgA",
                            "maneuver": {
                                "bearing_after": 33,
                                "location": [
                                    13.411491,
                                    52.527859
                                ],
                                "bearing_before": 0,
                                "type": "depart"
                            },
                            "duration": 5.3,
                            "distance": 45.1,
                            "name": "Zolastra\u00dfe",
                            "mode": "driving"
                        },
                        {
                            "intersections": [
                                {
                                    "out": 0,
                                    "location": [
                                        13.411851,
                                        52.5282
                                    ],
                                    "bearings": [
                                        120,
                                        210,
                                        300
                                    ],
                                    "entry": [
                                        true,
                                        false,
                                        false
                                    ],
                                    "in": 1
                                },
                                {
                                    "out": 1,
                                    "location": [
                                        13.415411,
                                        52.52692
                                    ],
                                    "bearings": [
                                        30,
                                        135,
                                        210,
                                        315
                                    ],
                                    "entry": [
                                        false,
                                        true,
                                        true,
                                        false
                                    ],
                                    "in": 3
                                }
                            ],
                            "geometry": "glr_IaozpARgAfBwJF]DOh@yAHSf@aADIR_@^m@",
                            "maneuver": {
                                "bearing_after": 114,
                                "type": "end of road",
                                "modifier": "right",
                                "bearing_before": 33,
                                "location": [
                                    13.411851,
                                    52.5282
                                ]
                            },
                            "duration": 23.7,
                            "distance": 306.9,
                            "name": "Torstra\u00dfe",
                            "mode": "driving"
                        },
                        {
                            "intersections": [
                                {
                                    "out": 1,
                                    "location": [
                                        13.415644,
                                        52.526756
                                    ],
                                    "bearings": [
                                        30,
                                        135,
                                        210,
                                        315
                                    ],
                                    "entry": [
                                        true,
                                        true,
                                        false,
                                        false
                                    ],
                                    "in": 3
                                },
                                {
                                    "out": 0,
                                    "location": [
                                        13.417204,
                                        52.52606
                                    ],
                                    "bearings": [
                                        120,
                                        210,
                                        300
                                    ],
                                    "entry": [
                                        true,
                                        true,
                                        false
                                    ],
                                    "in": 2
                                },
                                {
                                    "out": 1,
                                    "location": [
                                        13.419577,
                                        52.525172
                                    ],
                                    "bearings": [
                                        30,
                                        135,
                                        210,
                                        300
                                    ],
                                    "entry": [
                                        false,
                                        true,
                                        true,
                                        false
                                    ],
                                    "in": 3
                                },
                                {
                                    "out": 1,
                                    "location": [
                                        13.419958,
                                        52.524993
                                    ],
                                    "bearings": [
                                        30,
                                        120,
                                        210,
                                        300
                                    ],
                                    "entry": [
                                        true,
                                        true,
                                        false,
                                        false
                                    ],
                                    "in": 3
                                },
                                {
                                    "out": 0,
                                    "location": [
                                        13.421344,
                                        52.524428
                                    ],
                                    "bearings": [
                                        120,
                                        210,
                                        300
                                    ],
                                    "entry": [
                                        true,
                                        true,
                                        false
                                    ],
                                    "in": 2
                                }
                            ],
                            "geometry": "gcr_Iwf{pAN]Rg@Zy@X}@d@yABIFUh@oBJ_@Nq@Ps@DQRq@Ng@^sADKHYb@kALa@@CT{@Ri@Z_AJ[Lk@\\}A",
                            "maneuver": {
                                "bearing_after": 130,
                                "type": "new name",
                                "modifier": "straight",
                                "bearing_before": 139,
                                "location": [
                                    13.415644,
                                    52.526756
                                ]
                            },
                            "duration": 39.1,
                            "distance": 500.4,
                            "name": "Mollstra\u00dfe",
                            "mode": "driving"
                        },
                        {
                            "intersections": [
                                {
                                    "out": 0,
                                    "location": [
                                        13.421806,
                                        52.524283
                                    ],
                                    "bearings": [
                                        120,
                                        210,
                                        300
                                    ],
                                    "entry": [
                                        true,
                                        true,
                                        false
                                    ],
                                    "in": 2
                                }
                            ],
                            "geometry": "wsq_Iim|pAl@sCp@uCH_@He@H]BKJg@",
                            "maneuver": {
                                "bearing_after": 117,
                                "type": "continue",
                                "modifier": "straight",
                                "bearing_before": 117,
                                "location": [
                                    13.421806,
                                    52.524283
                                ]
                            },
                            "duration": 14,
                            "distance": 171.7,
                            "name": "Mollstra\u00dfe",
                            "mode": "driving"
                        },
                        {
                            "intersections": [
                                {
                                    "out": 0,
                                    "location": [
                                        13.424059,
                                        52.523574
                                    ],
                                    "bearings": [
                                        120,
                                        210,
                                        300
                                    ],
                                    "entry": [
                                        true,
                                        true,
                                        false
                                    ],
                                    "in": 2
                                }
                            ],
                            "geometry": "ioq_Ik{|pATkA",
                            "maneuver": {
                                "bearing_after": 116,
                                "type": "continue",
                                "modifier": "straight",
                                "bearing_before": 116,
                                "location": [
                                    13.424059,
                                    52.523574
                                ]
                            },
                            "duration": 2,
                            "distance": 28.6,
                            "name": "Mollstra\u00dfe",
                            "mode": "driving"
                        },
                        {
                            "intersections": [
                                {
                                    "in": 0,
                                    "entry": [
                                        true
                                    ],
                                    "bearings": [
                                        296
                                    ],
                                    "location": [
                                        13.424439,
                                        52.523461
                                    ]
                                }
                            ],
                            "geometry": "snq_Iw}|pA",
                            "maneuver": {
                                "bearing_after": 0,
                                "location": [
                                    13.424059,
                                    52.523574
                                ],
                                "bearing_before": 116,
                                "type": "arrive"
                            },
                            "duration": 0,
                            "distance": 0,
                            "name": "Mollstra\u00dfe",
                            "mode": "driving"
                        }
                    ],
                    "summary": "Torstra\u00dfe, Mollstra\u00dfe",
                    "duration": 84.1,
                    "distance": 1052.7
                }
            ],
            "duration": 84.1,
            "distance": 1052.7
        }
    ],
    "waypoints": [
        {
            "hint": "u-EAgL7hAIDoFAAADQAAAC4AAAAAAAAAAAAAAD3cAQA83AEApAAAAKOkzADzgiEDkqTMAPqCIQMAAAEBht76Hw==",
            "name": "Zolastra\u00dfe",
            "location": [
                13.411491,
                52.527859
            ]
        },
        {
            "hint": "CzABgP___39-BQAAFAAAADMAAAAAAAAANgAAABdJAQATSQEApAAAADfXzADFcSEDHdfMAKRxIQMAAAEBht76Hw==",
            "name": "Mollstra\u00dfe",
            "location": [
                13.424439,
                52.523461
            ]
        }
    ]
}
```

@MoKob is this a regression in the instruction collapsing?

**Comment by MoKob** [comment]
This is/was based on invalid discovery methods for roads where names are the same for multiple options.

**Comment by TheMarex** [comment]
Addressed by #2444.
